Mobl_ios_Callisto-master/MealTrackingPilot/LogCommonMeal/Views/OccasionPicker.swift

```diff
@@ -1,59 +1,44 @@
-//  all copies or substantial portions of the Software.
-//
-//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
-//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
-//  THE SOFTWARE.
 
+import Foundation
 import UIKit
 
 // MARK: - Protocols
 
-@objc public protocol PickerViewDataSource: class {
-    func pickerViewNumberOfItems(_ pickerView: PickerView) -> Int
-    func pickerView(_ pickerView: PickerView, titleForItem item: Int, index: Int) -> String
+@objc public protocol OccasionPickerDataSource: class {
+    func occasionPickerNumberOfItems(_ occasionPicker: OccasionPicker) -> Int
+    func occasionPicker(_ occasionPicker: OccasionPicker, titleForItem item: Int, index: Int) -> String
 }
 
-@objc public protocol PickerViewDelegate: class {
-    func pickerViewSpanForItems(_ pickerView: PickerView) -> CGFloat
-    @objc optional func pickerView(_ pickerView: PickerView, didSelectItem item: Int, index: Int)
-    @objc optional func pickerView(_ pickerView: PickerView, didTapItem item: Int, index: Int)
-    @objc optional func pickerView(_ pickerView: PickerView, styleForLabel label: UILabel, highlighted: Bool)
-    @objc optional func pickerView(_ pickerView: PickerView, viewForItem item: Int, index: Int, highlighted: Bool, reusingView view: UIView?) -> UIView?
-
-    @objc optional func pickerViewWillBeginMoving(_ pickerView: PickerView)
-    @objc optional func pickerViewDidEndMoving(_ pickerView: PickerView)
+@objc public protocol OccasionPickerDelegate: class {
+    func occasionPickerSpanForItems(_ occasionPicker: OccasionPicker) -> CGFloat
+    @objc optional func occasionPicker(_ occasionPicker: OccasionPicker, didSelectItem item: Int, index: Int)
+    @objc optional func occasionPicker(_ occasionPicker: OccasionPicker, didTapItem item: Int, index: Int)
+    @objc optional func occasionPicker(_ occasionPicker: OccasionPicker, styleForLabel label: UILabel, highlighted: Bool)
+    @objc optional func occasionPicker(_ occasionPicker: OccasionPicker, viewForItem item: Int, index: Int, highlighted: Bool, reusingView view: UIView?) -> UIView?
+    
+    @objc optional func occasionPickerWillBeginMoving(_ occasionPicker: OccasionPicker)
+    @objc optional func occasionPickerDidEndMoving(_ occasionPicker: OccasionPicker)
 }
 
-open class PickerView: UIView {
-
-    // MARK: Nested Types
-
-    fileprivate class SimplePickerCollectionViewCell: UICollectionViewCell {
+open class OccasionPicker: UIView {
+    
+    // MARK: UI elements
+    
+    fileprivate class OccasionPickerCollectionViewCell: UICollectionViewCell {
```

Comment:
Create a seperate file for OccasionPickerCollectionViewCell

Reply:
Done